vessim/sil.py

```diff
@@ -141,32 +162,37 @@ def start(self, microgrid: Microgrid) -> None:
 
         Thread(target=self._collect_set_requests_loop, daemon=True).start()
 
-    def step(self, time: datetime, p_delta: float, actor_infos: dict) -> None:
+    def step(self, time: datetime, p_delta: float, e_delta: float, actor_infos: dict) -> None:
         assert self.microgrid is not None
-        self.broker._add_microgrid_data(
-            time,
-            {
-                "microgrid": self.microgrid,
-                "actor_infos": actor_infos,
-                "p_delta": p_delta,
-            },
+        self.data_pipe_in.send(
+            (
+                time,
+                {
+                    "microgrid": self.microgrid,
+                    "actor_infos": actor_infos,
+                    "p_delta": p_delta,
+                    "e_delta": e_delta,
+                },
+            )
         )
 
-    def finalize(self) -> None:
-        self.broker._finalize()
-
     def _collect_set_requests_loop(self):
         while True:
+            start_time = time.time()
```

Comment:
```suggestion
            start_time = time.monotonic()
```
Using the system time for this is susceptible to changes to the system clock :I At worst, this could cause this thread to sleep for quite some time.

Reply:
Did not know that, thx!